skimage/feature/blob.py

```diff
@@ -91,15 +100,28 @@ def _prune_blobs(blobs_array, overlap):
 
     # iterating again might eliminate more blobs, but one iteration suffices
     # for most cases
-    for blob1, blob2 in itt.combinations(blobs_array, 2):
-        if _blob_overlap(blob1, blob2) > overlap:
-            if blob1[2] > blob2[2]:
-                blob2[2] = -1
-            else:
-                blob1[2] = -1
+    if len(blobs_array) == 0:
+        return np.array([])
+    sigma = blobs_array[:, -1].max()
+    distance = 2 * sigma * sqrt(blobs_array.shape[1] - 1)
+    try:
+        tree = spatial.cKDTree(blobs_array[:, :-1])
+        pairs = np.array(list(tree.query_pairs(distance)))
+    except AttributeError:  # scipy 0.9, min requirements
+        tree = spatial.KDTree(blobs_array[:, :-1])
+        pairs = np.array(list(tree.query_pairs(distance)))
+    if len(pairs) == 0:
+        return blobs_array
+    else:
+        for (i, j) in pairs:
+            blob1, blob2 = blobs_array[i], blobs_array[j]
+            if _blob_overlap(blob1, blob2) > overlap:
+                if blob1[-1] > blob2[-1]:
+                    blob2[-1] = -1
+                else:
+                    blob1[-1] = -1
```

Comment:
Can we just set to 0? -1 is ugly, and 0 does the job and is intuitive (making the blob "vanish").